raiden/network/proxies/token_network.py

```diff
@@ -194,56 +219,113 @@ def _new_netting_channel(self, partner: typing.Address, settle_timeout: int):
+                to_checksum_address(participant2),
+            )
+        assert isinstance(channel_identifier, typing.T_ChannelID)
+        if channel_identifier == 0:
+            raise ChannelIncorrectStateError(
+                f'When calling {called_by_fn} either 0 value was given for the '
+                'channel_identifier or getChannelIdentifier returned 0, meaning '
+                f'no channel currently exists between {pex(participant1)} and '
+                f'{pex(participant2)}',
+            )
+        return channel_identifier
+
+    def channel_exists_and_not_settled(
+            self,
+            participant1: typing.Address,
+            participant2: typing.Address,
+            channel_identifier: typing.ChannelID = None,
+    ) -> bool:
+        """Returns if the channel exists and is in a non-settled state"""
+        try:
+            channel_data = self.detail_channel(participant1, participant2, channel_identifier)
+        except ChannelIncorrectStateError:
+            return False
 
-        if not isinstance(channel_data['state'], typing.T_ChannelState):
+        if not isinstance(channel_data.state, typing.T_ChannelState):
             raise ValueError('channel state must be of type ChannelState')
 
         log.debug('channel data {}'.format(channel_data))
 
-        return channel_data['state'] > CHANNEL_STATE_NONEXISTENT
+        exists_and_not_settled = (
+            channel_data.state > ChannelState.NONEXISTENT and
+            channel_data.state < ChannelState.SETTLED
+        )
+        return exists_and_not_settled
 
     def detail_participant(
             self,
-            participant1: typing.Address,
-            participant2: typing.Address,
-    ) -> Dict:
+            channel_identifier: typing.ChannelID,
+            participant: typing.Address,
+            partner: typing.Address,
+    ) -> ParticipantDetails:
         """ Returns a dictionary with the channel participant information. """
+
         data = self._call_and_check_result(
             'getChannelParticipantInfo',
-            to_checksum_address(participant1),
-            to_checksum_address(participant2),
+            channel_identifier,
+            to_checksum_address(participant),
+            to_checksum_address(partner),
+        )
+        return ParticipantDetails(
+            address=participant,
+            deposit=data[ParticipantInfoIndex.DEPOSIT],
+            withdrawn=data[ParticipantInfoIndex.WITHDRAWN],
+            is_closer=data[ParticipantInfoIndex.IS_CLOSER],
+            balance_hash=data[ParticipantInfoIndex.BALANCE_HASH],
+            nonce=data[ParticipantInfoIndex.NONCE],
+        )
+
+    def detail_channel(
+            self,
+            participant1: typing.Address,
+            participant2: typing.Address,
+            channel_identifier: typing.ChannelID = None,
+    ) -> ChannelData:
+        """ Returns a dictionary with the channel specific information.
```

Comment:
Returns an instance of `ChannelData` could be a more accurate description

Reply:
Aha ... I see .. now .. your comment was on the wrong line. You mean to edit the docstrings. Will do.

Reply:
Yea, Github doesn't let you add comments on un-modified lines.